pkg/cmd/kind/load/docker-image/docker-image.go

```diff
@@ -121,38 +126,41 @@ func runE(logger log.Logger, flags *flagpole, args []string) error {
 
 	// pick only the nodes that don't have the image
 	selectedNodes := []nodes.Node{}
-	for _, node := range candidateNodes {
-		id, err := nodeutils.ImageID(node, imageName)
-		if err != nil || id != imageID {
-			selectedNodes = append(selectedNodes, node)
-			logger.V(0).Infof("Image: %q with ID %q not yet present on node %q, loading...", imageName, imageID, node.String())
+	fns := []func() error{}
+	for i, imageName := range imageNames {
+		imageID := imageIDs[i]
+		for _, node := range candidateNodes {
+			id, err := nodeutils.ImageID(node, imageName)
+			if err != nil || id != imageID {
+				selectedNodes = append(selectedNodes, node)
+				logger.V(0).Infof("Image: %q with ID %q not yet present on node %q, loading...", imageName, imageID, node.String())
+			}
+		}
+		if len(selectedNodes) == 0 {
+			return nil
```

Comment:
I think you should `continue` not return here, one image can be present in all nodes, but the next image maybe not

Reply:
i.e kind load `docker-image busybox,alpine,httpd`, if busybox is present in all nodes you return and alpine and httpd are not loaded

Reply:
oh right, nice catch. That's true. My bad. Missed to test on a multiple node-setup

Reply:
made the changes.